internal/collector/collector.go

```diff
@@ -249,12 +249,12 @@ func (f *Factory) ReconcileInputServices(er record.EventRecorder, k8sClient clie
 	for _, input := range f.ForwarderSpec.Inputs {
 		var listenPort int32
 		serviceName := f.ResourceNames.GenerateInputServiceName(input.Name)
-		if input.Receiver != nil && input.Receiver.ReceiverTypeSpec != nil {
-			if logging.IsHttpReceiver(&input) {
-				listenPort = input.Receiver.HTTP.Port
+		if input.Receiver != nil {
```

Comment:
Following up on the need for the functions... this is prime usage for a 'switch' on the type

Reply:
It definitively could, but we have changed so many times how do we know whether is something an HttpReceiver or a SyslogReceiver that I want to be a little defensive here. I think it is manageable with two receivers without a switch. May be a good candidate for a switch if the number of receiver types grow.